pokemon-checker/src/DataTransferObjects/MoveDTO.ts

```diff
@@ -1,31 +1,35 @@
 
-enum moveType {
+export enum damageClass {
   PHYSICAL = "physical",
   SPECIAL = "special",
-  STATUS = "status"
+  STATUS = "status",
+  NULL = "null"
 }
 
-type MoveConstructorOptions = {
+export type MoveConstructorOptions = {
   name: string,
-  damage: number,
-  accuracy: number,
-  type: moveType,
-  description: string
+  power?: number,
+  accuracy?: number,
+  damage_class?: damageClass,
+  description?: string,
+  url?: string
 }
 
-class MoveDTO {
+export class MoveDTO {
   public name: string;
-  public damage: number;
+  public power: number;
   public accuracy: number;
-  public type: moveType;
+  public damage_class: damageClass;
   public description: string;
+  public url: string;
 
   constructor(MoveConstructorOptions: MoveConstructorOptions) {
     this.name = MoveConstructorOptions.name;
-    this.damage = MoveConstructorOptions.damage;
-    this.accuracy = MoveConstructorOptions.accuracy;
-    this.type = MoveConstructorOptions.type;
-    this.description = MoveConstructorOptions.description;
+    this.power = MoveConstructorOptions.power ? MoveConstructorOptions.power : -1;
+    this.accuracy = MoveConstructorOptions.accuracy ? MoveConstructorOptions.accuracy : -1;
+    this.damage_class = MoveConstructorOptions.damage_class ? MoveConstructorOptions.damage_class : damageClass.NULL;
+    this.description = MoveConstructorOptions.description ? MoveConstructorOptions.description : "";
+    this.url = MoveConstructorOptions.url ? MoveConstructorOptions.url : "";
```

Comment:
I think we will need PP variable (I think it can potentially be optional if "Struggle" is a move since it technically has no PP stat)

Reply:
I'll stub a comment out for it, completely forgot while I was doing the pass over the interface.